### PR TITLE
ARROW-10455: [Rust] [CI] Fixed error in caching files

### PR DESCRIPTION
**Pull request by jorgecarleitao**
This PR fixes two errors:

1. `hashFiles` was not picking any file, thereby not differentiating cache keys.
2. `hashFiles` was not considering `Cargo.toml`, an important file in rust (like `package.json` in npm or `setup.py` in Python)

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         uses: actions/cache@v1
         with:
           path: .docker
-          key: debian-10-rust-${{ hashFiles('rust/**.rs') }}
+          key: debian-10-rust-${{ hashFiles('rust/**/**.rs', 'rust/**/Cargo.toml') }}
```
Comment (kszucs) [inline .github/workflows/rust.yml:71]: According to [GHA docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) `**` should match nested directories as well.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -118,8 +118,8 @@ jobs:
         uses: actions/cache@v1
         with:
           path: rust/target
-          key: windows-rust-${{ hashFiles('rust/**.rs') }}
-          restore-keys: windows-rust-
+          key: windows-rust-v1-${{ hashFiles('rust/**/**.rs', 'rust/**/Cargo.toml') }}
```
Comment (kszucs) [inline .github/workflows/rust.yml:121]: Is `v1` used here for invalidating the previous cache?

**Review by kszucs — APPROVED** [review]
LGTM, just a couple of nits.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (jorgecarleitao) [reply .github/workflows/rust.yml:71]: Without this, `hashFiles` becomes an empty string.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (jorgecarleitao) [reply .github/workflows/rust.yml:121]: yes, purely for that. Maybe there is a way to address this in another way, I was unable to make it work.

**Review by nevi-me — APPROVED** [review]

**Comment by alamb** [comment]
@jorgecarleitao  / @nevi-me  / @kszucs is there any outstanding work for this PR or shall we merge it in?

**Comment by kszucs** [comment]
@alamb nope, it's good to go.

**Review by kszucs — APPROVED** [review]

**Pull request closed without merging**